lib/backup/azremote/azblob.go

```diff
@@ -58,32 +65,64 @@ func (fs *FS) Init() error {
 		fs.Dir += "/"
 	}
 
+	domain := "blob.core.windows.net"
+	if storageDomain, ok := fs.env.LookupEnv(envStorageDomain); ok {
+		logger.Infof("Overriding default Azure blob domain with %q", storageDomain)
+		domain = storageDomain
+	}
+
+	connString, hasConnString := fs.env.LookupEnv(envStorageAccCs)
+	accountName, hasAccountName := fs.env.LookupEnv(envStorageAcctName)
+	accountKey, hasAccountKey := fs.env.LookupEnv(envStorageAccKey)
+	useDefault, _ := fs.env.LookupEnv(envStorageDefault)
+
 	var sc *service.Client
 	var err error
-	if cs, ok := envtemplate.LookupEnv(envStorageAccCs); ok {
-		sc, err = service.NewClientFromConnectionString(cs, nil)
+
+	// not used if connection string is set
+	serviceURL := fmt.Sprintf("https://%s.%s/", accountName, domain)
+
+	switch {
+	// can't specify any combination of more than one credential
+	case moreThanOne(hasConnString, (hasAccountName && hasAccountKey), (useDefault == "true" && hasAccountName)):
+		return fmt.Errorf("only one of connection string, account name and key, or default credential can be specified")
+	case hasConnString:
+		logger.Infof("Creating AZBlob service client from connection string")
+		sc, err = service.NewClientFromConnectionString(connString, nil)
 		if err != nil {
 			return fmt.Errorf("failed to create AZBlob service client from connection string: %w", err)
 		}
-	}
-
-	accountName, ok1 := envtemplate.LookupEnv(envStorageAcctName)
-	accountKey, ok2 := envtemplate.LookupEnv(envStorageAccKey)
-	if ok1 && ok2 {
+	case hasAccountName && hasAccountKey:
```

Comment:
Show this handle a case when only `envStorageAcctName` is set?
It is possible that user supplied only account name and it seems like this case will not return an error.

Reply:
it will return an error - i added a case to show that here: 92ab76fe5